lua/legendary/ui/init.lua

```diff
@@ -81,12 +82,21 @@ local function select_inner(opts, context, itemlist)
       return
     end
 
+    if opts.itemgroup then
+      State.most_recent_igroups[opts.itemgroup] = selected
+    else
+      State.most_recent_item = selected
+    end
```

Comment:
We need to track it both places in order for "repeat last execution" to work properly.

```suggestion
    if opts.itemgroup then
      State.itemgroup_history[opts.itemgroup] = selected
    end

    State.most_recent_item = selected
```

Reply:
Actually. Just thought of an issue with this approach. This will mean that if you select an item inside a group, the top-level most recent item gets lost. I think what we need to do is assign an ID to the top-level item list as well, and track it in the list like the others, then the `State.most_recent_item` can be used just for "re-execute last".